File: pymks_2_abaqus/array_test.py
import numpy as np
from scipy.io import savemat



sample_size = 3
n_samples = 1 * [sample_size]
arr = np.zeros((3, 21, 21, 21))
for i in range(0, 10):
    arr[:, i*2 + 1, i, i] = 1

#dataset_flattened = np.transpose(arr, (1, 2, 3, 0)).astype(bool)
#mat_dict = {'data': dataset_flattened.reshape(-1, sum(n_samples))}
mat_dict = {'data': arr}
savemat('data_test.mat', mat_dict)

File: pymks_2_abaqus/array_test.m
load('data_test.mat');
M=data;
N=zeros(3,21,21,21);
for i =1:10
    
    N(:,i*2,i,i) = 1;
end
N = permute(N,[2,3,4,1]);
N = reshape(N,21*21*21,3);
size(M)
size(N)

diff = abs(M(1,:,:,:)-N(1,:,:,:));
sum(sum(sum(sum(diff))))
